core/testing/include/Kiso_Testing.h

```diff
@@ -0,0 +1,192 @@
+{
+    uint8_t type;
+    uint8_t length;
+    char *value;
+} TlvElt_T;
+
+/**
+ *  @brief          Encapsulates the elements composing a message..
+ */
+typedef struct CCMsg_S
+{
+    MsgHdr_T header; /**< Header of the message */
+    uint8_t payload[TEST_MAX_PAYLOAD_LENGTH];
+    TlvElt_T tlvArray[CCHANNEL_MAX_NUMBER_OF_TLV_ELEMENTS]; /**< Parsed TLV array (from receive buffer) */
+    uint8_t payloadIndex;
+    uint8_t tlvArrayIndex;
+    uint8_t numberOfTlvElements; /**< Number of TLV elements for the message */
+    uint8_t rebootCounter;
+    bool isFree : 1; /**< This bit indicates whether the received message is processed by test runner or not */
+} CCMsg_T;
+
+/**
+ * @brief           Defines a prototype type for the setup functions of the test suites and test cases.
+ */
+typedef Retcode_T (*SetupFct_T)(CCMsg_T *ccmsg);
+
+/**
+ * @brief           Defines a prototype type for the tear down functions of the test suites and test cases.
+ */
+typedef Retcode_T (*TearDownFct_T)(CCMsg_T *ccmsg);
+
+/**
+ * @brief           Sefines a prototype type for the run functions of the test cases.
+ */
+typedef void (*RunFct_T)(CCMsg_T *ccmsg);
+
+/*###################### EXPORTED FUNCTIONS PROTOTYPES ###############################################################*/
+
+/**
+ * @brief           Initializes the Testing Framework.
+ * @details         this
+ * @param[in]       eId Id of the Test Entry
+ * @return          RETCODE_OK if initialized successfully error code otherwise
+ */
+Retcode_T Tests_Initialize(uint8_t eId, SetupFct_T setup, TearDownFct_T teardown);
+
+/**
+ * @brief           Registers a Test Suite
+ * @details         todo mak explain what the function does currently
+ * @param[in]       sId The identifier of the TestSuite to register
+ * @param[in]       setup A reference to the setup function of the TestSuite
+ * @param[in]       teardown A reference to the tear down function of the TestSuite
+ * @note            setup and tear down functions pointers can be null if nothing has to be done.
+ * @return          RETCODE_OK in case of success, error code otherwise.
+ */
+Retcode_T Tests_RegisterTestSuite(uint8_t sId, SetupFct_T setup, TearDownFct_T teardown);
+
+/**
+ * @brief           Registers a Test Case
+ * @details         todo mak explain what the function does 
+ * @param[in]       sId identifier of the Test Suite the test to register
+ * @param[in]       cId identifier of the Test Case to register
+ * @param[in]       setup A reference to the setup function of the Test Suite
+ * @param[in]       run A reference to the run function of the Test Suite
+ * @param[in]       teardown A reference to the tear down function of the Test Suite
+ * @note            setup and tear down functions pointers can be null if nothing has to be done.
+ * @return          RETCODE_OK in case of success, error code otherwise.
+ */
+Retcode_T Tests_RegisterTestCase(uint8_t sId, uint8_t cId, SetupFct_T setup, RunFct_T run, TearDownFct_T teardown);
```

Comment:
same here: why don't you name it testCaseId?  

tearDown might  better be camel cased